pandas/core/indexing.py

```diff
@@ -611,6 +615,42 @@ def _get_setitem_indexer(self, key):
                 raise
             raise IndexingError(key) from e
 
+    def _ensure_listlike_indexer(self, key, axis=None):
+        """
+        Ensure that a list-like of column labels are all present by adding them if
+        they do not already exist.
+
+        Parameters
+        ----------
+        key : _LocIndexer key or list-like of column labels
+            Target labels.
+        axis : key axis if known
```

Comment:
it looks like axis isnt used here, is it necessary?

Reply:
See https://github.com/pandas-dev/pandas/pull/29334#issuecomment-586245280